Cargo.toml

```diff
@@ -1,5 +1,5 @@
 [package]
-name = "sp1-bn254-poseidon2"
+name = "sp1-poseidon2-bn256"
```

Comment:
we should rename the repo to this IMHO. This matches the upstream convention

Reply:
cc @S1nus you would be the one to do this if you agree, else we should rename the crate & update downstream